src/Graph.ts

```diff
@@ -0,0 +1,174 @@
+/**
+ * Transform array to graph
+ */
+
+/**
+ * TYPES
+ */
+
+type node = {
+  x: Number;
+  y: Number;
+  value: Number;
+  isExit: Boolean;
+};
+
+type edge = {
+  from: string;
+  to: string;
+  direction: string;
+};
+
+type Graph = {
+  nodes: {
+    [coordinates: string]: node;
+  };
+  edges: Array<edge>;
+};
+
+/**
+ * FUNCTIONS
+ */
+
+export function boardToGraph(board: Array<Array<Number>>): Graph | undefined {
+
+  if (board.length < 1) return undefined;
+
+  let graphFromBoard: Graph = newBlankGraph();
+
+
+  const verticalLines: Number = board.length;
+    const horizontalLines: Number = board[0].length;
+  
+
+    for (let hIndex = 0; hIndex < horizontalLines; hIndex++){
+        for (let vIndex = 0; vIndex < verticalLines; vIndex++){
+
+            graphFromBoard = addBoardBoundaries(graphFromBoard, board, hIndex, vIndex)
+            graphFromBoard = addDirections(graphFromBoard, board, hIndex, vIndex)
+
+        }
+    }
+
+  return graphFromBoard;
+
+}
+
+function newBlankGraph(): Graph {
+  return {
+    nodes: {},
+    edges: [],
+  };
+}
+
+function addBoardBoundaries(graph: Graph, board: Array<Array<Number>>, hIndex: number, vIndex: number): Graph {
+
+    graph.nodes[`${hIndex},${vIndex}`] = {
+        x: hIndex,
+        y: vIndex,
+        value: board[vIndex][hIndex],
+        isExit: isAnExit(board, hIndex, vIndex)
+    }
+
+  return graph;
```

Comment:
This is not good: you're updating one of the arguments AND returning it. So your function may look like an immutable function but in fact it's mutable. 

My 2c: never mutate arguments, return a new board instead.

Reply:
The argument has been made immutable in the [last commit](https://github.com/marmelab/kuba-integration/commit/52c48e332abaf5397cf6debd1512e2111062ba31)